srml/offences/src/lib.rs

```diff
@@ -113,44 +108,18 @@ where
 		Self::deposit_event(Event::Offence(O::ID, time_slot.encode()));
 
 		let offenders_count = concurrent_offenders.len() as u32;
-		let previous_offenders_count = offenders_count - new_offenders.len() as u32;
 
 		// The amount new offenders are slashed
 		let new_fraction = O::slash_fraction(offenders_count, validator_set_count);
 
-		// The amount previous offenders are slashed additionally.
-		//
-		// Since they were slashed in the past, we slash by:
-		// x = (new - prev) / (1 - prev)
-		// because:
-		// Y = X * (1 - prev)
-		// Z = Y * (1 - x)
-		// Z = X * (1 - new)
-		let old_fraction = if previous_offenders_count > 0 {
-			let previous_fraction = O::slash_fraction(
-				offenders_count.saturating_sub(previous_offenders_count),
-				validator_set_count,
-			);
-			let numerator = new_fraction.saturating_sub(previous_fraction);
-			let denominator = Perbill::one().saturating_sub(previous_fraction);
-			denominator.saturating_mul(numerator)
-		} else {
-			new_fraction.clone()
-		};
+		let slash_perbill: Vec<_> = (0..concurrent_offenders.len())
+			.map(|_| new_fraction.clone()).collect();
 
-		// calculate how much to slash
-		let slash_perbill = concurrent_offenders
-			.iter()
-			.map(|details| {
-				if previous_offenders_count > 0 && new_offenders.contains(&details.offender) {
-					new_fraction.clone()
-				} else {
-					old_fraction.clone()
-				}
-			})
-			.collect::<Vec<_>>();
-
-		T::OnOffenceHandler::on_offence(&concurrent_offenders, &slash_perbill);
+		T::OnOffenceHandler::on_offence(
+			&concurrent_offenders,
+			&slash_perbill,
```

Comment:
nit: from the way that `slash_perbill ` is being created, I don't see why it has to be a collection rather than a single value. Seems to be always the same?